hs_core/hydroshare/utils.py

```diff
@@ -398,6 +385,7 @@ def replicate_resource_bag_to_user_zone(user, res_id):
     # needs to check whether res_id collection exists before getting/setting AVU on it to
     # accommodate the case where the very same resource gets deleted by another request when
     # it is getting downloaded
+    # TODO: why would we want to do anything at all if the resource does not exist???
```

Comment:
@alvacouch I agree, you can change to exit out without continuing if the resource does not exist, although this is just a precaution since it is called from a resource landing page and the resource should exist at this point. But in the future when this is exposed to REST API, there could be possibility the resource does not exist.